### PR TITLE
BIT-64 Add "open in github" button

### PR DESCRIPTION
**Pull request by mmoczkowski**

### CODE REVIEW

**Comment by maciej-wichrowski** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          26 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">app/src/main/java/io/stanwood/bitrise/data/model/RepoProvider.kt: New Kotlin class isn’t in kotlin source directory!</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">app/src/main/java/io/stanwood/bitrise/ui/builds/ui/BuildsFragment.kt:62: No space following if - please also correct in unchanged code!</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">app/src/main/java/io/stanwood/bitrise/ui/builds/ui/BuildsFragment.kt:70: No space following if - please also correct in unchanged code!</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">app/src/main/java/io/stanwood/bitrise/ui/builds/vm/BuildsViewModel.kt:96: No space following if - please also correct in unchanged code!</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">app/src/main/java/io/stanwood/bitrise/ui/dashboard/ui/DashboardFragment.kt:49: No space following if - please also correct in unchanged code!</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">app/src/main/java/io/stanwood/bitrise/ui/dashboard/ui/DashboardFragment.kt:62: No space following if - please also correct in unchanged code!</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">app/src/main/res/values/strings.xml:2: Translation contains linebreak</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">app/src/main/res/values/strings.xml:3: Translation contains linebreak</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">app/src/main/res/values/strings.xml:4: Translation contains linebreak</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">app/src/main/res/values/strings.xml:5: Translation contains linebreak</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">app/src/main/res/values/strings.xml:6: Translation contains linebreak</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">app/src/main/res/values/strings.xml:7: Translation contains linebreak</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">app/src/main/res/values/strings.xml:8: Translation contains linebreak</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">app/src/main/res/values/strings.xml:9: Translation contains linebreak</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">app/src/main/res/values/strings.xml:10: Translation contains linebreak</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">app/src/main/res/values/strings.xml:11: Translation contains linebreak</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">app/src/main/res/values/strings.xml:12: Translation contains linebreak</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">app/src/main/res/values/strings.xml:13: Translation contains linebreak</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">app/src/main/res/values/strings.xml:14: Translation contains linebreak</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">app/src/main/res/values/strings.xml:15: Translation contains linebreak</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">app/src/main/res/values/strings.xml:16: Translation contains linebreak</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">app/src/main/res/values/strings.xml:17: Translation contains linebreak</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">app/src/main/res/values/strings.xml:18: Translation contains linebreak</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">app/src/main/res/values/strings.xml:19: Translation contains linebreak</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">app/src/main/res/values/strings.xml:20: Translation contains linebreak</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">app/src/main/res/values/strings.xml:21: Translation contains linebreak</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: app/src/main/java/io/stanwood/bitrise/data/model/RepoProvider.kt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package io.stanwood.bitrise.data.model
+
+import com.google.gson.annotations.SerializedName
+
+enum class RepoProvider {
+    @SerializedName("github")
```
Comment (ubuntudroid) [inline app/src/main/java/io/stanwood/bitrise/data/model/RepoProvider.kt]: So there is only github? What about gitlab etc?

File: app/src/main/java/io/stanwood/bitrise/ui/builds/ui/BuildsFragment.kt [hunk 2]
```diff
@@ -44,4 +54,36 @@ class BuildsFragment : Fragment() {
             lifecycle.addObserver(viewModel)
             vm = viewModel
         }.root
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setHasOptionsMenu(true)
+        val currentActivity = activity
+        if (currentActivity is AppCompatActivity) {
+            toolbar.let {
```
Comment (ubuntudroid) [inline app/src/main/java/io/stanwood/bitrise/ui/builds/ui/BuildsFragment.kt]: Isn't the `let` here a bit superficial?

File: app/src/main/java/io/stanwood/bitrise/ui/builds/ui/BuildsFragment.kt [hunk 3]
```diff
@@ -44,4 +54,36 @@ class BuildsFragment : Fragment() {
             lifecycle.addObserver(viewModel)
             vm = viewModel
         }.root
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setHasOptionsMenu(true)
+        val currentActivity = activity
```
Comment (ubuntudroid) [inline app/src/main/java/io/stanwood/bitrise/ui/builds/ui/BuildsFragment.kt]: Why that val? You could also use `activity.let` if you must.

File: app/src/main/java/io/stanwood/bitrise/ui/builds/ui/BuildsFragment.kt [hunk 4]
```diff
@@ -44,4 +54,36 @@ class BuildsFragment : Fragment() {
...
+        if (currentActivity is AppCompatActivity) {
+            toolbar.let {
+                currentActivity.setSupportActionBar(it)
+            }
+        }
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu?, inflater: MenuInflater?) {
+        if (viewModel.isProvidedByGithub) {
+            inflater?.inflate(R.menu.builds, menu)
```
Comment (ubuntudroid) [inline app/src/main/java/io/stanwood/bitrise/ui/builds/ui/BuildsFragment.kt]: I'd always inflate the option menu and just enable/disable the github item for consistency.

File: app/src/main/java/io/stanwood/bitrise/ui/builds/vm/BuildsViewModel.kt [hunk 5]
```diff
@@ -53,6 +54,16 @@ class BuildsViewModel(private val router: NavController,
     val title: String
         get() = app.title
 
+    val isProvidedByGithub: Boolean
```
Comment (ubuntudroid) [inline app/src/main/java/io/stanwood/bitrise/ui/builds/vm/BuildsViewModel.kt]: Do you prefer using this as opposed to `fun isProvidedByGithub() = app.provider == RepoProvider.GITHUB`? If so, why? Just being curious.

File: app/src/main/java/io/stanwood/bitrise/ui/dashboard/ui/DashboardFragment.kt [hunk 6]
```diff
@@ -46,8 +46,8 @@ class DashboardFragment: Fragment() {
         super.onViewCreated(view, savedInstanceState)
         setHasOptionsMenu(true)
         val currentActivity = activity
-        if(currentActivity is AppCompatActivity) {
-            view.findViewById<Toolbar>(R.id.toolbar)?.let {
+        if (currentActivity is AppCompatActivity) {
```
Comment (ubuntudroid) [inline app/src/main/java/io/stanwood/bitrise/ui/dashboard/ui/DashboardFragment.kt]: See `currentActivity` remark before.

File: app/src/main/java/io/stanwood/bitrise/ui/builds/vm/BuildsViewModel.kt (hunk 5 above)
Comment (mmoczkowski) [reply app/src/main/java/io/stanwood/bitrise/ui/builds/vm/BuildsViewModel.kt]: If a method/function has no argument and returns something else than Unit I always replace it with property

File: app/src/main/java/io/stanwood/bitrise/data/model/RepoProvider.kt (hunk 1 above)
Comment (mmoczkowski) [reply app/src/main/java/io/stanwood/bitrise/data/model/RepoProvider.kt]: 1. I don't know all the providers supported by Bitrise
2. Will add more providers later

Comment (ubuntudroid) [reply app/src/main/java/io/stanwood/bitrise/data/model/RepoProvider.kt]: Will it crash when something else arrives here?

**Pull request merged**